### ISSUE TITLE
Train with pretrained weights question

### ISSUE THREAD

**Issue by wumy27** [zylo117/Yet-Another-EfficientDet-Pytorch#366]
I have transformed the custom dataset to coco2017 form and use your pre-trained model d2 to train my model.. But run like this:

![image](https://user-images.githubusercontent.com/46855459/83579726-a9150700-a56c-11ea-9800-2c10ee439d10.png)

My training process:
![image](https://user-images.githubusercontent.com/46855459/83579859-001adc00-a56d-11ea-9067-0437f6b1ea2a.png)

Can any one can give me some help?

**Comment by zylo117** [comment]
the objects are too small, you should try larger network or smaller anchors to fit you gt boxes

**Comment by wumy27** [comment]
> the objects are too small, you should try larger network or smaller anchors to fit you gt boxes

Thanks for your reply. You mean when i run your *efficientdet_test.py*, I should use smaller anchors?

**Comment by wumy27** [comment]
> the objects are too small, you should try larger network or smaller anchors to fit you gt boxes

Thanks for your reply. I will use smaller anchors and see if it can help.

**Comment by oukohou** [comment]
@wumy27 same problem when using ` --head_only True`, what's the result after you using smaller anchors?

**Comment by zylo117** [comment]
It has nothing to do will `head_only`, you should try more suitable anchors or larger network/ larger image size

**Comment by oukohou** [comment]
@zylo117 after seeing other similar issues, maybe the problem lies on the category id? my categories are like following:
```python
"categories": [
    {
      "id": 2,
      "supercategory": "none",
      "name": "car"
    },
    {
      "id": 1,
      "supercategory": "none",
      "name": "bicycle"
    },
    {
      "id": 5,
      "supercategory": "none",
      "name": "tricycle"
    },
    {
      "id": 3,
      "supercategory": "none",
      "name": "truck"
    },
    {
      "id": 6,
      "supercategory": "none",
      "name": "electrocycle"
    },
    {
      "id": 4,
      "supercategory": "none",
      "name": "crowd"
    },
    {
      "id": 0,
      "supercategory": "none",
      "name": "motorcycle"
    }
```
which starts from `0`, while you said [`category id starts from 1`](https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch/issues/373#issuecomment-640046890)  
I think this caused my model to predict really bad?

**Comment by zylo117** [comment]
it has to start from 1

**Comment by oukohou** [comment]
@zylo117 yeah，thanks, I'll change the category id. And as the regression loss if far smaller than cls loss, I manually add weights to regression loss, which is :
`loss = cls_loss*class_loss_weight + reg_loss*reg_loss_weight`  
do you consider it reasonable?

**Comment by zylo117** [comment]
You are welcome to try and feel free to share the result.
Reg loss is much lower than cls loss becasue regression is much easier to train than classification.

**Comment by dthaicuong** [comment]
> it has to start from 1.
> And if the pictures are all green, it's probably under fitting. you can visualize the loss graph on tensorboard to get more infomation.

I tried few times without category_id starting from 1. My cat_id starts from 0 and it works pretty good.
@oukohou: I met blue images as well. It's from ori_imgs somehow. But I just extract ROIS after invert_affine and plot them onto img = cv2.imread()

**Comment by zylo117** [comment]
@dthaicuong did you modify the code to adapt to cat_id starting from 0? I've been training well with cat id starting from 1 for months.

**Comment by dthaicuong** [comment]
> modify the code to adapt to

@zylo117 I did not change anything from your repo. For few first tries, I forgot "__background__" class as usual COCO but results are pretty good (or caused by low diversity in my dataset)
